### DISCUSSION TITLE
Can't get outgoing Email to work

### DISCUSSION THREAD

**Discussion by SRFurst** [jgyates/genmon#1091] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

v1.19.02

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

Have Genmon working flawlessly on a 22kW air cooled with Pushover notifications. But I cannot seem to configure outgoing Email correctly. I am using smtp.comcast.net (not Gmail) as my smtp server, using the same settings (port 465, SSL enabled) as my Email program. When I try to send a test email I get, "An error occurred: Please run connect() first". Genserv.log says the same thing (genserv.py:726)

Any suggestions are appreciated.

**Comment by jgyates** [comment]
You did not include how you have your mail settings configured on the settings page.

I would try port 587 and make sure the other settings below the port number are in the "off" position. This is the most secure and most widely implemented approach. Comcast should support 587 and possibly 465 for older clients. 

<img width="546" alt="Screenshot 2024-03-26 at 7 26 05 PM" src="https://github.com/jgyates/genmon/assets/4513286/6869982d-bf81-40b4-95bc-a74f55a5af94">

The above recommendation would make the settings the same for gmail and comcast (except for the SMTP server address). This is the same for hotmail / outlook as well.

Most mail servers implement SSL on port 465 and TSL on 587 so if you want to try port 465 then the most used server setup would require you to set "Disable TLS" to the on position and "Use SSL" to the on position.

Let me know how it goes. If it does not work then please provide a picture of your settings.

**Reply by SRFurst** [reply]
A copy of my /etc/mymail.conf is below. These are the settings I use in my Email program (Port 465, SSL on) so I know they are correct. I also tried using some other SMTP servers (smtp.mail.me.com, etc) with the same settings and got a different error. I then tried your suggestion (turning all encryption off and using port 587) for Comcast which did work. I worry (slightly) about my user/password being sent in clear text, tho...

-------------------

[MyMail]

# optionally disable email use by the program. set to true to disable all email functionality
disableemail = false

# optionally disable receiving mail
disableimap = true

# optionally disable sending mail
disablesmtp = false

# The password for the email account used for sending and receiving email
email_pw = <redacted> 

# the email account
email_account = <redacted>@comcast.net

# address that email are sent to
email_recipient = <redacted>@comcast.net

# the email from which the emails are sent (sender email address)
sender_account = <redacted>@comcast.net

# optional, name of the sender, i.e. Genmon or John Doe
sender_name = Genmon

# the SMTP mail server used for sending email, leave empty to disable SMTP, i.e. "smtp_server ="
smtp_server = smtp.comcast.net

# the IMAP server address used for receiving email, leave empty to disable IMAP, i.e. "imap_server ="
imap_server =

# the SMTP port used on the SMTP server for sending email
# Note about ports and email security: typically port 587 is used for
#  TLS and port 465 is used for SSL
smtp_port = 465

# use SSL encryption to send emails
ssl_enabled = true

# do not use TLS encryption if true, the default value is false (use TLS).
# This is parameter ignored if ssl_enable is true. Enabling this and disabling
# SSL encryption will result in no SMTP encryption.
tls_disable = false

# optionally disable smtp authentication disabling this will result in no SMTP authentication
smtpauth_disable = false

# use BCC instead of To for sending mail
usebcc = False

# extend wait time in seconds on errors
extend_wait = 0
<redacted>@comcast.net = outage,error,warn

**Reply by jgyates** [reply, marked as answer]
It looks like you are using a TLS port but have both SSL and TLS enabled in the settings. Most mail severs only use SSL only on port 
 465 and only TLS on port 587. Follow the recommendations above and let me know. Also, I do not recommend editing the config file directly, use the GUI as it has more error checking to prevent invalid entries.

**Reply by SRFurst** [reply]
Thank you, disabling TLS while using port 465 did the trick.

**Reply by jgyates** [reply]
Great! I am glad is it working for you.
